### PR TITLE
Update to latest SDK

### PR DESCRIPTION
**Pull request by jkotalik**
Was trying to resolve an issue with nuget restore locally, didn't work for me unfortunately. However I already fixed all issues that came in from the new SDK so might as well make a PR for it.

### CODE REVIEW

File: src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj [hunk 1]
```diff
@@ -3,13 +3,7 @@
...
     <Description>Razor is a markup syntax for adding server-side logic to web pages. This assembly contains infrastructure supporting Razor MSBuild integration.</Description>
 
-    <!--
-      This is intentionally targeting netcoreapp3.0. This allows using the Microsoft.NET.Sdk.Razor package without having to solve diamond dependency problems in AspNetCore.
-      We use a runtimeconfig.json.template to allow roll forwards.
-      In addition, the SDK stamps an exact version of the shared fx to use in all runtimeconfig.json that are contained in the SDK.
-      See the discussion here: https://github.com/dotnet/aspnetcore-internal/issues/3201#issuecomment-539631557
-    -->
-    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <TargetFramework>$(DefaultTargetFramework)</TargetFramework>
```
Comment (dougbu) [inline src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj:6]: This is a big shift. @NTaylorMullen and @pranavkm do we really want to move this to work only when targeting 5.0❔ The alternative would be to switch to `netcoreapp3.1`.

**Comment by dougbu** [comment]
@jkotalik what needs this slightly-newer SDK❔ I ask because it doesn't seem compatible with our projects and dependencies and I'm not aware of an immediate need for this SDK.
``` text
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202: Package Newtonsoft.Json 12.0.2 is not compatible with net (.NETFramework,Version=v0.0). Package Newtonsoft.Json 12.0.2 supports: [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - net20 (.NETFramework,Version=v2.0) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - net35 (.NETFramework,Version=v3.5) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - net40 (.NETFramework,Version=v4.0) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - net45 (.NETFramework,Version=v4.5) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - netstandard1.0 (.NETStandard,Version=v1.0) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - netstandard1.3 (.NETStandard,Version=v1.3) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - netstandard2.0 (.NETStandard,Version=v2.0) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - portable-net40+sl5+win8+wp8+wpa81 (.NETPortable,Version=v0.0,Profile=Profile328) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202:   - portable-net45+win8+wp8+wpa81 (.NETPortable,Version=v0.0,Profile=Profile259) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1201: Project Microsoft.CodeAnalysis.Razor is not compatible with net (.NETFramework,Version=v0.0). Project Microsoft.CodeAnalysis.Razor supports: netstandard2.0 (.NETStandard,Version=v2.0) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
/Users/runner/work/1/s/src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj : error NU1202: Package Microsoft.Css.Parser 1.0.0-20200708.1 is not compatible with net (.NETFramework,Version=v0.0). Package Microsoft.Css.Parser 1.0.0-20200708.1 supports: netstandard2.0 (.NETStandard,Version=v2.0) [/Users/runner/work/1/s/.dotnet/sdk/5.0.100-rc.1.20407.13/NuGet.targets]
```

And, no, I don't know how the SDK comes up w/ `.NETFramework,Version=v0.0`.

File: src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj (hunk 1 above)
Comment (jkotalik) [reply src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj:6]: I already confirmed with @pranavkm

Comment (dougbu) [reply src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj:6]: Well, @NTaylorMullen said `netcoreapp3.1` in an email earlier today&hellip;

**Comment by jkotalik** [comment]
> @jkotalik what needs this slightly-newer SDK❔ I ask because it doesn't seem compatible with our projects and dependencies and I'm not aware of an immediate need for this SDK.

I think the variable DefaultFrameworkVersion wasn't available in that project. Setting to netcoreapp3.1 for now.

And why not update to a slightly newer SDK? I don't see the hesitation? This found a few bugs we had in our code through new analyzers.

File: src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj [hunk 2]
```diff
@@ -3,13 +3,7 @@
...
     <Description>Razor is a markup syntax for adding server-side logic to web pages. This assembly contains infrastructure supporting Razor MSBuild integration.</Description>
 
-    <!--
-      This is intentionally targeting netcoreapp3.0. This allows using the Microsoft.NET.Sdk.Razor package without having to solve diamond dependency problems in AspNetCore.
-      We use a runtimeconfig.json.template to allow roll forwards.
-      In addition, the SDK stamps an exact version of the shared fx to use in all runtimeconfig.json that are contained in the SDK.
-      See the discussion here: https://github.com/dotnet/aspnetcore-internal/issues/3201#issuecomment-539631557
-    -->
-    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
[the remark this answers is not in the document]
Comment (pranavkm) [reply src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj:6]: You'll have to update the paths in Razor.Sdk.csproj: https://github.com/dotnet/aspnetcore/pull/24371/files#diff-8f9947c64e80be276e8abd868edc59fbR72

**Comment by dougbu** [comment]
> I don't see the hesitation?

My hesitation is due to the newer SDK seeming not to work. If you can get it working, fine. (We're already pretty far away from the "North star" of using public preview versions of the SDK.)

File: src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj (hunk 1 above)
Comment (NTaylorMullen) [reply src/Razor/Microsoft.AspNetCore.Razor.Tools/src/rzc.csproj]: Wouldn't net5.0 re-introduce the diamond dependency the comment above introduces?

**Comment by Tratcher** [comment]
> Was trying to resolve an issue with nuget restore locally, didn't work for me unfortunately. However I already fixed all issues that came in from the new SDK so might as well make a PR for it.

What restore issue? This one?
```
D:\github\AspNetCore\.dotnet\sdk\5.0.100-rc.1.20379.10\NuGet.targets(128,5): error : Unable to load the service index for source https://api.nuget.org/v3/index.json.
D:\github\AspNetCore\.dotnet\sdk\5.0.100-rc.1.20379.10\NuGet.targets(128,5): error :   The SSL connection could not be established, see inner exception.
D:\github\AspNetCore\.dotnet\sdk\5.0.100-rc.1.20379.10\NuGet.targets(128,5): error :    Received an unexpected EOF or 0 bytes from the transport stream.
    0 Warning(s)
    1 Error(s)
```

**Comment by Tratcher** [comment]
Ah https://github.com/NuGet/Home/issues/9893

**Pull request closed without merging**